pyrsia_node/src/network/behavior.rs

```diff
@@ -24,24 +27,45 @@ pub struct MyBehaviour {
     floodsub: Floodsub,
     kademlia: Kademlia<MemoryStore>,
     mdns: Mdns,
+    #[behaviour(ignore)]
+    response_sender: tokio::sync::mpsc::Sender<String>,
 }
 
 impl MyBehaviour {
-    pub fn new(floodsub: Floodsub, kademlia: Kademlia<MemoryStore>, mdns: Mdns) -> Self {
+    pub fn new(
+        floodsub: Floodsub,
+        kademlia: Kademlia<MemoryStore>,
+        mdns: Mdns,
+        response_sender: tokio::sync::mpsc::Sender<String>,
+    ) -> Self {
         MyBehaviour {
             floodsub,
             kademlia,
             mdns,
+            response_sender,
         }
     }
 
     pub fn floodsub_mut(&mut self) -> &mut Floodsub {
         &mut self.floodsub
     }
 
-    pub fn lookup_blob(&mut self, hash: String) -> Result<QueryId, Error> {
-        let num = std::num::NonZeroUsize::new(2).ok_or(Error::ValueTooLarge)?;
-        Ok(self.kademlia.get_record(&Key::new(&hash), Quorum::N(num)))
+    pub async fn lookup_blob(&mut self, hash: String) {
+        let num = std::num::NonZeroUsize::new(2)
+            .ok_or(Error::ValueTooLarge)
+            .unwrap();
+        self.kademlia.get_record(&Key::new(&hash), Quorum::N(num));
+    }
+
+    pub async fn list_peers(&mut self, peer_id: PeerId) {
+        self.kademlia.get_closest_peers(peer_id);
```

Comment:
so someone does not pass another node's ID

```suggestion
        self.kademlia.get_closest_peers(local_peer_id);
```

Reply:
"local_peer_id" is local to main class which has been passed to the behavior on KademliaEvent

Reply:
Maybe a small comment "get peers next to `peer_id` in DHT" so it's more clear? I made the mistake